agent/hibinit-agent

```diff
@@ -39,23 +39,42 @@ IMDS_API_TOKEN_PATH = 'latest/api/token'
 IMDS_SPOT_ACTION_PATH = 'latest/meta-data/hibernation/configured'
 MAX_SWAP_SIZE_OFFSET_ALLOWED = 100 * 1024
 
+# Sigterm_handler Behaviour Modifiers
+CRITICAL_PROCESS_IN_PROGRESS = False
+SHUTDOWN_REQUESTED = False
+
+
 def log(message):
     if log_to_syslog:
         syslog.syslog(message)
 
-def sigterm_handler(signal, frame):
-    #save the state here or do whatever you want
-    log('Process killed cleaning up!')
-    cmd = "swapon -s | cut -f1,1 | grep -w {name}"
-    cmd = cmd.format(name=SWAP_FILE)
-    swapoff = "swapoff {filename}"
-    cmd = check_output(cmd, shell=True)
-    if cmd.strip() == SWAP_FILE:
-        print("Turning off swap for cleanup")
-        swapoff = swapoff.format(filename=SWAP_FILE)
-        check_call(swapoff, shell=True)
-    if os.path.isfile(SWAP_FILE) and os.access(SWAP_FILE, os.R_OK):
-        os.remove(SWAP_FILE)
+
+def sigterm_handler(signum, frame):
+    if CRITICAL_PROCESS_IN_PROGRESS:
+        global SHUTDOWN_REQUESTED
+        SHUTDOWN_REQUESTED = True
```

Comment:
Just curious what is left to be done by the agent after the critical process. The reason I asked this is  as we keep adding more global variables , we might encounter some side effects at some point. if SHUTDOWN_REQUESTED is not required at this point, I think its better to remove it but I will leave it up to you

Reply:
Yes, agreed; The use of global variables are usually rather pungent code smell. 

In this case, this use was deemed necessary for sigterm_handler behaviour to be modified mid process.

The alternative was to implement an alternative handler (perhaps `sigterm_critical_handler`) to run when receiving a `SIGTERM` kill signal, then returning behaviour back to normal after said critical section is completed.

This may be something to revisit when we refactor this package in future, as there are many other improvements we can make as well.